pkg/rtc/participant.go

```diff
@@ -368,10 +372,14 @@ func (p *ParticipantImpl) SetPermission(permission *livekit.ParticipantPermissio
 	video.SetCanPublishData(permission.CanPublishData)
 	video.Hidden = permission.Hidden
 	video.Recorder = permission.Recorder
-	p.grants.Store(grants)
+
+	canPublish := video.GetCanPublish()
+
+	p.grants.Video = video
```

Comment:
This is a redundant re-assign as `Video` is a pointer and these should be modified in-place, but re-assigning it makes it easier to follow (I think)

Reply:
yeah.. you are right it's unnecessary.